elasticsearch_dsl/_async/document.py

```diff
@@ -317,9 +369,13 @@ async def update(
             doc_meta["if_seq_no"] = self.meta["seq_no"]
             doc_meta["if_primary_term"] = self.meta["primary_term"]
 
+        i = self._get_index(index)
+        assert i is not None
```

Comment:
This little dance is just to help mypy understand that the index is never going to be `None`.